docs/usage_and_configuration/the_basics.md

```diff
@@ -124,6 +124,148 @@ running `black --help`.
 
 </details>
 
+### Code input alternatives
+
+#### Standard Input
+
+_Black_ supports formatting code via stdin, with the result being printed to stdin. Just
+let _Black_ know with `-` as the path.
+
+```console
+$ echo "print ( 'hello, world' )" | black -
+print("hello, world")
+reformatted -
+All done! ✨ 🍰 ✨
+1 file reformatted.
+```
+
+**Tip:** if you need _Black_ to treat stdin input as a file passed directly via the CLI,
+use `--stdin-filename`. Useful to make sure _Black_ will respect the `--force-exclude`
+option on some editors that rely on using stdin.
+
+#### As a string
+
+You can also pass code as a string using the `-c` / `--code` option.
+
+```console
+$ black --code "print ( 'hello, world' )"
+print("hello, world")
+```
+
+```{warning}
+--check, --diff, and --safe / --fast have no effect when using -c / --code. Safety
+checks normally turned on by default that verify _Black_'s output are disabled as well.
```

Comment:
Maybe mention that this is considered a bug and we'll eventually change it. I think there's an issue about it.